tests/test_bqm.py

```diff
@@ -23,18 +23,28 @@
 import numpy as np
 
 
-class TestFixedShapeAPI:
+class TestBQMAPI:
     """
     Tests for BQMs like AdjArrayBQM (doesn't try to change the shape)
     """
 
+    def test__len__(self):
+        bqm = self.BQM(np.ones((107, 107)))
+        self.assertEqual(len(bqm), 107)
+
     def test_construction_symmetric(self):
         bqm = self.BQM(np.ones((5, 5)))
         for u, v in itertools.combinations(range(5), 2):
             self.assertEqual(bqm.get_quadratic(u, v), 2)  # added
         for u in range(5):
             self.assertEqual(bqm.get_linear(u), 1)
 
+    # def test_construction_nans(self):
+    #     bqm = self.BQM(np.nan((5, 5)))
+    #     self.assertEqual(bqm.shape, (5, 0))
+    #     for u in range(5):
+    #         self.assertEqual(bqm.get_linear(u), 0)
+
```

Comment:
My opinion on how to treat nans is to raise an error. There isn't an obvious way to interpret them and no established convention.